src/tests/FastBoard_unittest.cpp

```diff
@@ -0,0 +1,342 @@
+    b.set_square(4, 5, FastBoard::WHITE);
+    b.set_square(2, 2, FastBoard::BLACK);
+    b.set_square(4, 3, FastBoard::WHITE);
+    b.set_square(1, 2, FastBoard::BLACK);
+    b.set_square(6, 3, FastBoard::WHITE);
+    b.set_square(2, 3, FastBoard::BLACK);
+    b.set_square(5, 2, FastBoard::WHITE);
+    b.set_square(0, 0, FastBoard::BLACK);
+    b.set_square(6, 6, FastBoard::WHITE);
+    return b;
+}
+
+/*
+      a b c d e
+    5 . . O O .  5
+    4 . . O . O  4
+    3 O O O O .  3
+    2 . . O . .  2
+    1 . . O . .  1
+      a b c d e
+*/
+FastBoard create_5x5_all_white_field() {
+    FastBoard b;
+    b.reset_board(5);
+    b.set_square(1, 2, FastBoard::WHITE);
+    b.set_square(2, 1, FastBoard::WHITE);
+    b.set_square(2, 2, FastBoard::WHITE);
+    b.set_square(2, 3, FastBoard::WHITE);
+    b.set_square(2, 4, FastBoard::WHITE);
+    b.set_square(3, 2, FastBoard::WHITE);
+    b.set_square(3, 4, FastBoard::WHITE);
+    b.set_square(4, 3, FastBoard::WHITE);
+    b.set_square(0, 2, FastBoard::WHITE);
+    b.set_square(2, 0, FastBoard::WHITE);
+    return b;
+}
+
+
+TEST(FastBoardTest, Board3x3) {
+    FastBoard b;
+    b.reset_board(3);
+    const char *expected = "\n"
+        "   a b c \n"
+        " 3 . . .  3\n"
+        " 2 . . .  2\n"
+        " 1 . . .  1\n"
+        "   a b c \n\n";
+    
+    EXPECT_EQ(expected,  b.serialize_board());
+    EXPECT_EQ(3, b.get_boardsize());
+}
+
+TEST(FastBoardTest, MakeBlackMoveOn19x19) {
+    FastBoard b;
+    b.reset_board(19);
+    b.set_square(b.get_vertex(2, 1), FastBoard::BLACK);
+    
+    const char *expected = "\n"
+        "   a b c d e f g h j k l m n o p q r s t \n"
+        "19 . . . . . . . . . . . . . . . . . . . 19\n"
+        "18 . . . . . . . . . . . . . . . . . . . 18\n"
+        "17 . . . . . . . . . . . . . . . . . . . 17\n"
+        "16 . . . + . . . . . + . . . . . + . . . 16\n"
+        "15 . . . . . . . . . . . . . . . . . . . 15\n"
+        "14 . . . . . . . . . . . . . . . . . . . 14\n"
+        "13 . . . . . . . . . . . . . . . . . . . 13\n"
+        "12 . . . . . . . . . . . . . . . . . . . 12\n"
+        "11 . . . . . . . . . . . . . . . . . . . 11\n"
+        "10 . . . + . . . . . + . . . . . + . . . 10\n"
+        " 9 . . . . . . . . . . . . . . . . . . .  9\n"
+        " 8 . . . . . . . . . . . . . . . . . . .  8\n"
+        " 7 . . . . . . . . . . . . . . . . . . .  7\n"
+        " 6 . . . . . . . . . . . . . . . . . . .  6\n" 
+        " 5 . . . . . . . . . . . . . . . . . . .  5\n"
+        " 4 . . . + . . . . . + . . . . . + . . .  4\n"
+        " 3 . . . . . . . . . . . . . . . . . . .  3\n"
+        " 2 . . X . . . . . . . . . . . . . . . .  2\n"
+        " 1 . . . . . . . . . . . . . . . . . . .  1\n"
+        "   a b c d e f g h j k l m n o p q r s t \n\n";
+    EXPECT_EQ(expected, b.serialize_board());
+}
+
+TEST(FastBoardTest, GetVertexOn19x19) {
+    FastBoard b;
+    b.reset_board(19);
+    EXPECT_EQ(22, b.get_vertex(0, 0));
```

Comment:
Given that these vertex coordinates are internal to Leela, it's not clear to me what the use of this test is.

Wouldn't it make more sense to combine both tests and just see if you can round-trip the coordinates, rather than check if they are equal to arbitrary numbers?

Reply:
Yes, that is a good suggestion. The test now verifies that get_xy(get_vertex) is consistent.